airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -58,29 +59,33 @@ class AppflowBaseOperator(BaseOperator):
 
     ui_color = "#2bccbd"
 
+    template_fields = ("flow_name", "source", "source_field", "filter_date")
+
     UPDATE_PROPAGATION_TIME: int = 15
 
     def __init__(
         self,
-        source: str,
         flow_name: str,
         flow_update: bool,
+        source: str | None = None,
```

Comment:
This is breaking change, changing the parameter order can break many user DAGs

Reply:
though this is the base operator, so no one is supposed to use it outside of this file.

Reply:
But other operators have the same issue below. e.g. `AppflowRunOperator `. Usually we force using operator names in operators but I guess we missed this one :|

Reply:
Operators/Sensors `__init__` constructor use only keyword-arguments, even if user provide positional, than error would happen. So change order in operators it is pretty safe. Unfortunetly you can't do the same things in Hooks 😢 

https://github.com/apache/airflow/blob/41d9be072abacc47393f700aa8fb98bc2b9a3713/airflow/models/baseoperator.py#L389-L396

Reply:
If we want to keep the same order in Operator

```python
    def __init__(
        self,
        *, 
        # All parameters below are keyword-only, `keyword argument` != `optional argument`
        # Because Operators are keyword-only by design it is safe to do this trick
        source: str | None = None,
        flow_name: str,
        flow_update: bool,
```

Reply:
ah ok so I remembered well, and this is fine :D

Reply:
Nice!